### DISCUSSION TITLE
Keywords should also carry signature information

### DISCUSSION THREAD

**Discussion by snoe** [clj-kondo/clj-kondo#1235] [Ideas]
Suppose I have a re-frame app, it's actually quite helpful at "call" sites to know the signatures for things.
So in hooks we have `api/reg-keyword!`, it should be possible in that call to also add the signature keys that var-definitions have:
 
`:fixed-arities :varargs-min-arity :arglists-str`

For instance in the screenshot, clojure-lsp could look for these signature fields and provide information to the user about the "args"
<img width="997" alt="image" src="https://user-images.githubusercontent.com/105012/112564186-6db82300-8da0-11eb-9541-d40df4730c96.png">

**Comment by borkdude** [comment]
@snoe Maybe we can add an an `:analysis` key to the hook function so the `:analysis` data will just be included in the analysis? In this way users can add to the analysis whatever they want, maybe?

**Reply by snoe** [reply]
Makes sense to me.
